### PR TITLE
Meraki web auth

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Adds web authentication support with the Meraki access points
# Impacts

Another switch module
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Added support for web authentication on the Meraki cloud controller

### CODE REVIEW

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 1]
```diff
@@ -3524,6 +3524,32 @@ Hostapd (software)
...
 To configure Hostapd software you can use the same configuration parameters above in the configuration file.
 
+Meraki
+~~~~~~
+
+In this section, we will cover the configuration of the Meraki controller to use Web authentication.
+
+CAUTION: There is currently no way to reevaluate the access of the device using PacketFence. This means that there is no real-time action when trying to isolate a device. The RADIUS reauthentication delay will determine the maximum response time of the access reevaluation.
+
+First create a new SSID and then edit it's settings.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: its settings

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 2]
```diff
@@ -3524,6 +3524,32 @@ Hostapd (software)
...
+
+Configure it as seen in the screenshot below.
+
+image::docs/images/meraki-radius-config.png[scaledwidth="100%",alt="RADIUS configuration"]
+
+NOTE: Make sure to configure your RADIUS server so it's accessible externally under a domain name or a public IP address. Replace `pf.domain.net` by what you will configure. Refer to the Meraki documentation for more details.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Perhaps we could also say that only the ports XYZ need to be open.

File: lib/pf/Switch/Meraki/AP_http.pm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+pf::Switch::Meraki::AP_http
+
+=head1 SYNOPSIS
+
+The pf::Switch::Meraki::AP_http module implements an object oriented interface to 
+manage the external captive portal on Meraki access points
+
+=head1 STATUS
+
+Developed and tested on an MR12 access point
```
Comment (extrafu) [inline lib/pf/Switch/Meraki/AP_http.pm]: on a MR12

**Comment by extrafu** [comment]
@whitx Please test with the doc as-is. Also check with @julsemaan for the firewall rules, and @lzammit if the firewall needs a reboot.

**Comment by whitx** [comment]
Screenshots missing

**Comment by whitx** [comment]
Need to add a DHCP helper on the switch to make it work or have the AP on the same subnet than PacketFence. Meraki send a natted IP(of the client) to PF where it should just send the one that was attributed by any dhcp server to the client.
Try to use last firmware update but didn't change anything.

**Comment by julsemaan** [comment]
Someone has last thoughts about this PR before it gets merged ?

**Pull request merged**